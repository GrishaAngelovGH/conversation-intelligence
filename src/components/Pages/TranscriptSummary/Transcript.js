import PropTypes from "prop-types"

import Attendee from "./Attendee"

const TranscriptRow = ({ attendee, duration, content }) => (
  <div className="row border rounded mt-3 p-1 bg-light">
    <div className="col-md-12">
      <div className="row">
        <div className="col-md-12">
          <Attendee shortName={attendee.shortName} fullName={attendee.fullName} />

          <div className="row mt-1 align-items-center">
            <div className="col-md-2">
              {duration}
            </div>
            <div className="col-md-10">
              {content}
            </div>
          </div>
        </div>
      </div>
    </div>
  </div >
)

TranscriptRow.propTypes = {
  attendee: PropTypes.object.isRequired,
  duration: PropTypes.string.isRequired,
  content: PropTypes.element.isRequired
}

const Transcript = () => {

  const transcript = [
    { id: 1, attendee: { shortName: 'JS', fullName: 'James Smith' }, duration: '00:10', content: (<div>Good morning, everyone.</div>) },
    { id: 2, attendee: { shortName: 'JS', fullName: 'James Smith' }, duration: '00:15', content: (<div>Hi, <span className="fw-bold">Madison</span>! How are you?</div>) },
    { id: 3, attendee: { shortName: 'MB', fullName: 'Madison Butler' }, duration: '00:05', content: (<div>Very good, thanks !</div>) },
    { id: 4, attendee: { shortName: 'MB', fullName: 'Madison Butler' }, duration: '00:17', content: (<div>Oh, before I forget, can you please send me the <span className="fw-bold">quote</span> we discussed <span className="fw-bold">yesterday</span>?</div>) },
    { id: 5, attendee: { shortName: 'JS', fullName: 'James Smith' }, duration: '00:04', content: (<div>Sure, I will send you an email with the <span className="fw-bold">quote</span></div>) }
  ]

  return (
    <div className="row p-3">
      <div className="col-md-12">
        <h3 className="text-center">Transcript</h3>
        {
          transcript.map(v => (
            <TranscriptRow key={v.id} attendee={v.attendee} duration={v.duration} content={v.content} />
          ))
        }
      </div>
    </div>
  )
}

export default Transcript