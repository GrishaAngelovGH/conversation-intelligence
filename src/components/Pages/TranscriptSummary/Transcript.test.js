import { render } from '@testing-library/react'

import Transcript, { TranscriptRow } from './Transcript'

test('should render component', () => {
  const container = render(<Transcript />)

  expect(container).toMatchSnapshot()
})

test('should render TranscriptRow', () => {
  const transcript = { attendee: { shortName: 'JS', fullName: 'James Smith' }, duration: '00:10', content: (<div>Good morning, everyone.</div>) }

  const container = render(<TranscriptRow attendee={transcript.attendee} duration={transcript.duration} content={transcript.content} />)

  expect(container).toMatchSnapshot()
})