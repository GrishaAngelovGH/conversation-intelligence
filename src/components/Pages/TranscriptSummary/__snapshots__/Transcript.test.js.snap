// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TranscriptRow 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row border rounded mt-3 p-1 bg-light"
      >
        <div
          class="col-md-12"
        >
          <div
            class="row"
          >
            <div
              class="col-md-12"
            >
              <div
                class="row mt-1 align-items-center"
              >
                <div
                  class="col-md-2"
                >
                  <div
                    class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                    style="width: 40px; height: 40px;"
                  >
                    JS
                  </div>
                </div>
                <div
                  class="col-md-6"
                >
                  <h4
                    class="m-0"
                  >
                    James Smith
                  </h4>
                </div>
              </div>
              <div
                class="row mt-1 align-items-center"
              >
                <div
                  class="col-md-2"
                >
                  00:10
                </div>
                <div
                  class="col-md-10"
                >
                  <div>
                    Good morning, everyone.
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row border rounded mt-3 p-1 bg-light"
    >
      <div
        class="col-md-12"
      >
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="row mt-1 align-items-center"
            >
              <div
                class="col-md-2"
              >
                <div
                  class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                  style="width: 40px; height: 40px;"
                >
                  JS
                </div>
              </div>
              <div
                class="col-md-6"
              >
                <h4
                  class="m-0"
                >
                  James Smith
                </h4>
              </div>
            </div>
            <div
              class="row mt-1 align-items-center"
            >
              <div
                class="col-md-2"
              >
                00:10
              </div>
              <div
                class="col-md-10"
              >
                <div>
                  Good morning, everyone.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row p-3"
      >
        <div
          class="col-md-12"
        >
          <h3
            class="text-center"
          >
            Transcript
          </h3>
          <div
            class="row border rounded mt-3 p-1 bg-light"
          >
            <div
              class="col-md-12"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      <div
                        class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                        style="width: 40px; height: 40px;"
                      >
                        JS
                      </div>
                    </div>
                    <div
                      class="col-md-6"
                    >
                      <h4
                        class="m-0"
                      >
                        James Smith
                      </h4>
                    </div>
                  </div>
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      00:10
                    </div>
                    <div
                      class="col-md-10"
                    >
                      <div>
                        Good morning, everyone.
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row border rounded mt-3 p-1 bg-light"
          >
            <div
              class="col-md-12"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      <div
                        class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                        style="width: 40px; height: 40px;"
                      >
                        JS
                      </div>
                    </div>
                    <div
                      class="col-md-6"
                    >
                      <h4
                        class="m-0"
                      >
                        James Smith
                      </h4>
                    </div>
                  </div>
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      00:15
                    </div>
                    <div
                      class="col-md-10"
                    >
                      <div>
                        Hi, 
                        <span
                          class="fw-bold"
                        >
                          Madison
                        </span>
                        ! How are you?
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row border rounded mt-3 p-1 bg-light"
          >
            <div
              class="col-md-12"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      <div
                        class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                        style="width: 40px; height: 40px;"
                      >
                        MB
                      </div>
                    </div>
                    <div
                      class="col-md-6"
                    >
                      <h4
                        class="m-0"
                      >
                        Madison Butler
                      </h4>
                    </div>
                  </div>
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      00:05
                    </div>
                    <div
                      class="col-md-10"
                    >
                      <div>
                        Very good, thanks !
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row border rounded mt-3 p-1 bg-light"
          >
            <div
              class="col-md-12"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      <div
                        class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                        style="width: 40px; height: 40px;"
                      >
                        MB
                      </div>
                    </div>
                    <div
                      class="col-md-6"
                    >
                      <h4
                        class="m-0"
                      >
                        Madison Butler
                      </h4>
                    </div>
                  </div>
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      00:17
                    </div>
                    <div
                      class="col-md-10"
                    >
                      <div>
                        Oh, before I forget, can you please send me the 
                        <span
                          class="fw-bold"
                        >
                          quote
                        </span>
                         we discussed 
                        <span
                          class="fw-bold"
                        >
                          yesterday
                        </span>
                        ?
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row border rounded mt-3 p-1 bg-light"
          >
            <div
              class="col-md-12"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      <div
                        class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                        style="width: 40px; height: 40px;"
                      >
                        JS
                      </div>
                    </div>
                    <div
                      class="col-md-6"
                    >
                      <h4
                        class="m-0"
                      >
                        James Smith
                      </h4>
                    </div>
                  </div>
                  <div
                    class="row mt-1 align-items-center"
                  >
                    <div
                      class="col-md-2"
                    >
                      00:04
                    </div>
                    <div
                      class="col-md-10"
                    >
                      <div>
                        Sure, I will send you an email with the 
                        <span
                          class="fw-bold"
                        >
                          quote
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row p-3"
    >
      <div
        class="col-md-12"
      >
        <h3
          class="text-center"
        >
          Transcript
        </h3>
        <div
          class="row border rounded mt-3 p-1 bg-light"
        >
          <div
            class="col-md-12"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    <div
                      class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                      style="width: 40px; height: 40px;"
                    >
                      JS
                    </div>
                  </div>
                  <div
                    class="col-md-6"
                  >
                    <h4
                      class="m-0"
                    >
                      James Smith
                    </h4>
                  </div>
                </div>
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    00:10
                  </div>
                  <div
                    class="col-md-10"
                  >
                    <div>
                      Good morning, everyone.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row border rounded mt-3 p-1 bg-light"
        >
          <div
            class="col-md-12"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    <div
                      class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                      style="width: 40px; height: 40px;"
                    >
                      JS
                    </div>
                  </div>
                  <div
                    class="col-md-6"
                  >
                    <h4
                      class="m-0"
                    >
                      James Smith
                    </h4>
                  </div>
                </div>
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    00:15
                  </div>
                  <div
                    class="col-md-10"
                  >
                    <div>
                      Hi, 
                      <span
                        class="fw-bold"
                      >
                        Madison
                      </span>
                      ! How are you?
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row border rounded mt-3 p-1 bg-light"
        >
          <div
            class="col-md-12"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    <div
                      class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                      style="width: 40px; height: 40px;"
                    >
                      MB
                    </div>
                  </div>
                  <div
                    class="col-md-6"
                  >
                    <h4
                      class="m-0"
                    >
                      Madison Butler
                    </h4>
                  </div>
                </div>
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    00:05
                  </div>
                  <div
                    class="col-md-10"
                  >
                    <div>
                      Very good, thanks !
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row border rounded mt-3 p-1 bg-light"
        >
          <div
            class="col-md-12"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    <div
                      class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                      style="width: 40px; height: 40px;"
                    >
                      MB
                    </div>
                  </div>
                  <div
                    class="col-md-6"
                  >
                    <h4
                      class="m-0"
                    >
                      Madison Butler
                    </h4>
                  </div>
                </div>
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    00:17
                  </div>
                  <div
                    class="col-md-10"
                  >
                    <div>
                      Oh, before I forget, can you please send me the 
                      <span
                        class="fw-bold"
                      >
                        quote
                      </span>
                       we discussed 
                      <span
                        class="fw-bold"
                      >
                        yesterday
                      </span>
                      ?
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row border rounded mt-3 p-1 bg-light"
        >
          <div
            class="col-md-12"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    <div
                      class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                      style="width: 40px; height: 40px;"
                    >
                      JS
                    </div>
                  </div>
                  <div
                    class="col-md-6"
                  >
                    <h4
                      class="m-0"
                    >
                      James Smith
                    </h4>
                  </div>
                </div>
                <div
                  class="row mt-1 align-items-center"
                >
                  <div
                    class="col-md-2"
                  >
                    00:04
                  </div>
                  <div
                    class="col-md-10"
                  >
                    <div>
                      Sure, I will send you an email with the 
                      <span
                        class="fw-bold"
                      >
                        quote
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
